src/org/sosy_lab/cpachecker/cfa/parser/llvm/CFABuilder.java

```diff
@@ -1175,21 +1176,31 @@ private CRightHandSide getConstant(final Value pItem, final String pFileName)
 
   private CRightHandSide getConstant(final Value pItem, CType pExpectedType, final String pFileName)
       throws LLVMException {
+    assert pItem.isConstant() : "getConstant called on non-constant value";
     FileLocation location = getLocation(pItem, pFileName);
     if (pItem.isConstantInt()) {
       long constantValue = pItem.constIntGetSExtValue();
       return new CIntegerLiteralExpression(
           getLocation(pItem, pFileName), pExpectedType, BigInteger.valueOf(constantValue));
 
     } else if (pItem.isConstantPointerNull()) {
-      return new CPointerExpression(location, pExpectedType, getNull(location, pExpectedType));
+      return new CCastExpression(location, pExpectedType,
+                                 getNull(location, pExpectedType));
 
     } else if (pItem.isConstantExpr()) {
       return getExpression(pItem, pExpectedType, pFileName);
 
     } else if (pItem.isUndef()) {
       CType constantType = typeConverter.getCType(pItem.typeOf());
-      String undefName = "__VERIFIER_undef_" + constantType.toString().replace(' ', '_');
+      /* get the name of the type and sanitize it
+       * to form a correct C identifier */
+      String typeName = constantType.toString().replace(' ', '_');
+      typeName = typeName.replace('(', '_');
+      typeName = typeName.replace(')', '_');
+      typeName = typeName.replace(':', '_');
+      typeName = typeName.replace("*", "_ptr_");
```

Comment:
Actually, is this name unambiguous across multiple undefs?

Reply:
For different types, you get different names. Instead of `__VERIFIER_undef_(int : 8)*` you get `__VERIFIER_undef__int___8__ptr`. So if the type would be ` int : 8 ` (notice the spaces around) instead of `(int : 8)`, you would get the same names, but I do not think you can get such two types, right?

Reply:
Yeah I agree, sorry - not sure what I was thinking about when asking this ;-)